store/sql_channel_store.go

```diff
@@ -405,7 +405,7 @@ func (s SqlChannelStore) get(id string, master bool, allowFromCache bool) StoreC
 				if metrics != nil {
 					metrics.IncrementMemCacheHitCounter("Channel")
 				}
-				result.Data = cacheItem.(*model.Channel)
+				result.Data = (cacheItem.(*model.Channel)).DeepCopy()
```

Comment:
`updateChannel()` and various others mutates the returned object from the cache.